### PR TITLE
implement graphs

### PR DESCRIPTION
**Pull request by ericcf**
[#96588144]

### CODE REVIEW

File: src/js/caches/ResourceCache.js [hunk 1]
```diff
@@ -32,22 +32,27 @@
...
+            record.id === d.id) {
+          rawData[i] = record;
+          storage()[key] = JSON.stringify(rawData);
+          isRejected = true;
         }
+
+        if (isRejected) { return; }
       });
 
       if (!isRejected) {
```
Comment (eschlange) [inline src/js/caches/ResourceCache.js]: Looks like this conditional is no longer necessary.

Comment (ericcf) [inline src/js/caches/ResourceCache.js]: Good catch - thanks.

**Comment by eschlange** [comment]
@ericcf Aside from that small change, everything looks good. I'd say its ready to merge.

File: src/js/caches/ResourceCache.js (hunk 1 above)
Comment (ericcf) [inline src/js/caches/ResourceCache.js]: Actually I do need that conditional - the `return` in line 55 is returning from the anonymous function starting at line 42.

**Pull request merged**